src/test/java/com/connexta/store/controllers/StoreControllerIngestTests.java

```diff
@@ -0,0 +1,334 @@
+        "2012-W48-6", // ISO_WEEK_DATE
+        "20111203", // BASIC_ISO_DATE
+        "Sun, 11 Jun 2017 14:32:28 GMT", // RFC_1123_DATE_TIME
+        "2017-06-11T14:32:28.120+0000", // ISO_DATE_TIME missing offset :
+        "             ",
+        ""
+      })
+  public void testInvalidLastModifiedHeader(String badDate) throws Exception {
+    mockMvc
+        .perform(
+            multipart("/ingest")
+                .file(FILE)
+                .file(METACARD)
+                .header("Accept-Version", ACCEPT_VERSION)
+                .header(LAST_MODIFIED, badDate)
+                .param("correlationId", CORRELATION_ID))
+        .andExpect(status().isBadRequest());
+
+    verifyNoInteractions(mockStoreService);
+  }
+
+  /**
+   * TODO Test RuntimeException and Throwable thrown by {@link StoreService#ingest(Long, String,
+   * InputStream, String, Long, InputStream)}
+   */
+  @ParameterizedTest(name = "{0} is the response status code when IngestService#ingest throws {1}")
+  @MethodSource("exceptionThrownByIngestServiceAndExpectedResponseStatus")
+  public void testIngestServiceExceptions(
+      final Throwable throwable, final HttpStatus expectedResponseStatus) throws Exception {
+    doThrow(throwable).when(mockStoreService).ingest(any(), any(), any(), any(), any(), any());
+
+    mockMvc
+        .perform(
+            multipart("/ingest")
+                .file(FILE)
+                .file(METACARD)
+                .param("correlationId", CORRELATION_ID)
+                .header(LAST_MODIFIED, LAST_MODIFIED_DATE)
+                .header("Accept-Version", ACCEPT_VERSION))
+        .andExpect(status().is(expectedResponseStatus.value()));
+  }
+
+  private static Stream<Arguments> exceptionThrownByIngestServiceAndExpectedResponseStatus() {
+    return Stream.of(
+        Arguments.of(
+            new StoreException("Test", new Throwable()), HttpStatus.INTERNAL_SERVER_ERROR));
+  }
+
+  @NotNull
+  private static InputStream inputStreamContentEq(@NotNull final InputStream expected) {
+    return argThat(
+        actual -> {
+          try {
+            return IOUtils.contentEquals(expected, actual);
+          } catch (final IOException e) {
+            fail("Unable to compare input streams", e);
+            return false;
+          }
+        });
+  }
+
+  @ParameterizedTest
+  @NullAndEmptySource
+  @ValueSource(strings = " ")
+  public void testIngestBlankContentType(final String contentType) {
+    when(mockFile.getContentType()).thenReturn(contentType);
+    assertThat(
+        storeController
+            .ingest(
+                ACCEPT_VERSION,
+                OffsetDateTime.parse(LAST_MODIFIED_DATE),
+                mockFile,
+                CORRELATION_ID,
+                METACARD)
+            .getStatusCode(),
+        is(HttpStatus.ACCEPTED));
+  }
+
+  @ParameterizedTest
+  @NullAndEmptySource
+  @ValueSource(strings = "   ")
+  public void testBadFilename(final String filename) {
```

Comment:
❓ Why would these situations return 202 Accepted? Wouldn't we be returning something along the lines of 5XX?

Reply:
I think it would request a BAD REQUEST code because the filename in the request is not valid.

Reply:
Yeah, you're right. To clarify, my comment refers to all of the bad parameter tests at the bottom of the test class. It looks like they all assert Accepted.